features/steps/steps.py

```diff
@@ -352,6 +354,21 @@ def then_stdout_matches_regexp(context):
     assert_that(context.process.stdout.strip(), matches_regexp(context.text))
 
 
+@then("stdout is formatted as `{output_format}` and has keys")
+def then_stdout_is_formatted_and_has_keys(context, output_format):
+    output = context.process.stdout.strip()
+    if output_format == "json":
+        data = json.loads(output)
+    elif output_format == "yaml":
+        data = yaml.safe_load(output)
+
+    keys = context.text.split()
+    in_data = [k for k in keys if k in data]
+
+    assert_that(len(keys), equal_to(len(data)))
+    assert_that(len(keys), equal_to(len(in_data)))
+
```

Comment:
Does this error message needs to be improved as well? Right now, we have just the assertion error with the numbers. Should it be clearer on why, or we leave this to whoever runs it?

Reply:
We could potentially show which key is missing. This could provide useful information if the test fails

Reply:
Ok, so, I am currently verifying if the keys are exactly those listed on the test, no more, no less.
I can change it to point to missing keys, or extraneous ones on error.
Is that it, or we can pass on extraneous keys? (My guess is no, it should be exact the same)

Reply:
I think it should exactly the same